components/arm/universalrobots/ur5e_test.go

```diff
@@ -435,3 +435,76 @@ func TestArmReconnection(t *testing.T) {
 	childCancel()
 	_ = ua.Close(ctx)
 }
+
+func TestUpdateAction(t *testing.T) {
+	var remote atomic.Bool
+
+	remote.Store(false)
+
+	statusBlob, err := os.ReadFile("armBlob")
+	test.That(t, err, test.ShouldBeNil)
+
+	logger := golog.NewTestLogger(t)
+	parentCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	ctx, childCancel := context.WithCancel(parentCtx)
+
+	closer, _, _, err := setupListeners(ctx, statusBlob, &remote)
+	test.That(t, err, test.ShouldBeNil)
+
+	cfg := config.Component{
+		Name: "testarm",
+		ConvertedAttributes: &AttrConfig{
+			Speed:               0.3,
+			Host:                "localhost",
+			ArmHostedKinematics: false,
+		},
+	}
+
+	newCfg := config.Component{
+		Name: "testarm",
+		ConvertedAttributes: &AttrConfig{
+			Speed:               0.5,
+			Host:                "localhost",
+			ArmHostedKinematics: false,
+		},
+	}
+
+	lastCfg := config.Component{
+		Name: "testarm",
+		ConvertedAttributes: &AttrConfig{
+			Speed:               0.5,
+			Host:                "new",
+			ArmHostedKinematics: false,
+		},
+	}
+
+	injectRobot := &inject.Robot{}
+
+	arm, err := URArmConnect(parentCtx, injectRobot, cfg, logger)
```

Comment:
might not be possible but if there is a way to create the arm by just using the URArm struct that would make your code easier since you don't the arm communication emulation layer

Reply:
I do not think it is possible to do that, but I defer to @biotinker for confirmation.